File: Repositories/js/js-monolith-images/dataLoadModule.js
import fs from "fs";
import path from "path";

const dataDir = path.join(process.cwd(), "data");
const imageDir = path.join(dataDir, "images");

export const imageBuffers = {};
export const featureIndex = {};
export const metadataIndex = {};

function loadJson(file) {
    return JSON.parse(
        fs.readFileSync(path.join(dataDir, file), "utf-8")
    );
}


function loadImages() {
    const files = fs.readdirSync(imageDir);

    files.forEach(file => {
        const id = path.parse(file).name;
        imageBuffers[id] = fs.readFileSync(
            path.join(imageDir, file)
        );
    });
}

export async function loadData() {

    loadImages();

    const features = loadJson("features.json");
    const metadata = loadJson("metadata.json");

    features.forEach(f => {
        featureIndex[f.id] = f.vector;
    });

    metadata.forEach(m => {
        metadataIndex[m.id] = m;
    });

    console.log(
        `Loaded ${Object.keys(imageBuffers).length} images`
    );
}

File: Repositories/python/python-monolith-images/src/data_load_module.py
import json
import os

current_dir = os.path.dirname(os.path.abspath(__file__))
data_dir = os.path.join(current_dir, "..", "data")
image_dir = os.path.join(data_dir, "images")

image_buffers = {}
feature_index = {}
metadata_index = {}


def load_json(filename):
    with open(os.path.join(data_dir, filename), "r", encoding="utf-8") as f:
        return json.load(f)




def load_images():
    for file in os.listdir(image_dir):
        image_id = os.path.splitext(file)[0]
        with open(os.path.join(image_dir, file), "rb") as f:
            image_buffers[image_id] = f.read()


def load_data():
    load_images()

    for item in load_json("features.json"):
        feature_index[item["id"]] = item["vector"]

    for item in load_json("metadata.json"):
        metadata_index[item["id"]] = item

    print(
        f"[DATA] Loaded {len(image_buffers)} images, "
        f"{len(feature_index)} feature vectors, "
        f"{len(metadata_index)} metadata records"
    )


load_data()
